tests/spec/test_spec.py

```diff
@@ -16,13 +18,19 @@
 
 
 def pytest_generate_tests(metafunc):
```

Comment:
Comment and not a complaint: this method of generating test parameters is interesting and I had no idea pytest had this API.  Cheers on this one!

Reply:
This is the same pattern used in py-evm and seems to be appropriate and work well for situations like this where you want individual tests for some *unknown* amount of tests from an external source.

https://github.com/ethereum/py-evm/blob/100b8c7d5267a6a6c9ae9c0472963db217b390db/eth/tools/fixtures/generation.py#L43-L91

Also worth noting that it'd be possible to do this with a parametrized fixture which *might* be a little less magic but not by very much.